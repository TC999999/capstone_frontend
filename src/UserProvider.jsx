import UserContext from "./UserContext.js";
import { useState, useEffect } from "react";
import marketAPI from "../api.js";

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  //sets context for user for entire app
  const currentUser = (user) => {
    setUser(user);
  };

  // //On initial render, get user token from local storage, gets the current user from the api, and set as context
  useEffect(() => {
    async function getToken() {
      let token = localStorage.getItem("market-token");
      if (token) {
        let userInfo = await marketAPI.getCurrentUser();
        currentUser(userInfo);
      }
    }
    getToken();
  }, []);

  //These functions and user state will be used in context
  //Set the token in local storage, retrives the current user from the api, and sets the user as context on login
  const logIn = async (username, token) => {
    localStorage.setItem("market-token", token);
    let userInfo = await marketAPI.getUserInfo(username);
    currentUser(userInfo);
  };

  //remove token and user from local storage and set user context to null on logout
  const logOut = () => {
    localStorage.removeItem("market-token");
    currentUser(null);
  };

  //update current user in local storage and context
  const updateUser = async (username) => {
    let userInfo = await marketAPI.getUserInfo(username);
    currentUser(userInfo);
  };

  return (
    <UserContext.Provider
      value={{ user, currentUser, logIn, logOut, updateUser }}
    >
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
