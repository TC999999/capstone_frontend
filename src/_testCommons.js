const TEST_USER = {
  username: "u1",
  firstName: "fn1",
  lastName: "ln1",
  email: "testemail1@gmail.com",
  address: "100 address",
  zipCode: "11111",
  regionOrState: "state1",
  city: "city1",
  country: "country1",
  latitude: "10.99876",
  longitude: "-99.99981",
  isAdmin: false,
  isFlagged: false,
  items: [
    {
      itemID: 1,
      name: "i1",
      initialPrice: 100,
      imageURL: "test_url_1",
      condition: "great",
      description: "test item",
    },
    {
      itemID: 2,
      name: "i2",
      initialPrice: 100,
      imageURL: "test_url_2",
      condition: "great",
      description: "test item",
    },
  ],
  reviews: [],
  reports: [],
  pastPurchases: [
    {
      purchaseID: 1,
      itemName: "i4",
      sellerUser: "u3",
      imageURL: "test_url_4",
      price: 150,
      soldAt: expect.any(Date),
      fromFlaggedUser: false,
    },
  ],
};

const TEST_ITEM = {
  id: 1,
  name: "i1",
  imageURL: "test_url_1",
  initialPrice: 100,
  condition: "great",
  description: "test item",
  sellerUser: "u1",
  isSold: false,
  types: [{ id: 1, name: "electronics" }],
  location: {
    city: "city1",
    address: "100 address",
    latitude: "10.99876",
    longitude: "-99.99981",
    regionOrState: "state1",
    country: "country1",
    zipCode: "11111",
  },
};

const TEST_ITEMS = [
  {
    id: 3,
    name: "i3",
    imageURL: "test_url_3",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u2",
    types: [
      { id: 1, name: "electronics" },
      { id: 3, name: "books" },
    ],
    location: {
      city: "city2",
      regionOrState: "state2",
      country: "country2",
      zipCode: "22222",
    },
  },
  {
    id: 2,
    name: "i2",
    imageURL: "test_url_2",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u1",
    types: [
      { id: 1, name: "electronics" },
      { id: 2, name: "movies" },
    ],
    location: {
      city: "city1",
      regionOrState: "state1",
      country: "country1",
      zipCode: "11111",
    },
  },
  {
    id: 1,
    name: "i1",
    imageURL: "test_url_1",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u1",
    types: [{ id: 1, name: "electronics" }],
    location: {
      city: "city1",
      regionOrState: "state1",
      country: "country1",
      zipCode: "11111",
    },
  },
];

const TEST_MESSAGE = {
  id: 3,
  from: "u2",
  to: "u1",
  itemID: 1,
  itemName: "i3",
  body: "test body",
  createdAt: "2024-07-09 14:23:43.469126-04",
};

const TEST_MESSAGE_2 = {
  id: 3,
  from_username: "u2",
  to_username: "u1",
  itemID: 1,
  item_name: "i3",
  body: "test body",
  sent_at: "2024-07-09 14:23:43.469126-04",
};

const TEST_MESSAGES = {
  conversation: [
    {
      id: 2,
      from_username: "u3",
      to_username: "u1",
      item_name: "i4",
      itemID: 4,
      body: "test body",
      sent_at: "2024-07-09 14:23:43.469126-04",
    },
    {
      id: 1,
      from_username: "u1",
      to_username: "u3",
      item_name: "i4",
      itemID: 4,
      body: "test body",
      sent_at: "2024-07-09 14:23:43.469126-04",
    },
  ],
};

const TEST_REVIEW = {
  id: 1,
  reviewed: "u3",
  reviewer: "u1",
  rating: 10,
  body: "test review",
  madeAt: "2024-07-09 14:23:43.469126-04",
};

const TEST_REPORT = {
  id: 1,
  reportedUser: "u2",
  reporterUser: "u1",
  body: "test body",
  madeAt: "2024-07-09 14:23:43.469126-04",
  isCleared: false,
};

const TEST_REPORTS = [
  {
    id: 1,
    reportedUser: "u2",
    reporterUser: "u1",
    body: "test body",
    madeAt: "2024-07-09 14:23:43.469126-04",
    isCleared: false,
  },
  {
    id: 2,
    reportedUser: "u1",
    reporterUser: "u2",
    body: "test body",
    madeAt: "2024-07-09 14:23:43.469126-04",
    isCleared: false,
  },
];

const TEST_PURCHASE = {
  purchaseID: 1,
  itemName: "i4",
  sellerUser: "u3",
  imageURL: "test_url_4",
  price: 150,
  soldAt: "2024-07-09 14:23:43.469126-04",
  fromFlaggedUser: false,
};

module.exports = {
  TEST_USER,
  TEST_ITEM,
  TEST_ITEMS,
  TEST_MESSAGE,
  TEST_MESSAGE_2,
  TEST_MESSAGES,
  TEST_REPORT,
  TEST_REPORTS,
  TEST_REVIEW,
  TEST_PURCHASE,
};
