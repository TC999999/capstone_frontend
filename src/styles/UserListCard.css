.user-list-card {
  border: solid 2px red;
  border-radius: 10px;
  background-color: rgb(150, 242, 242);
  box-shadow: 5px 5px 5px black;
  padding: 5px;
  margin: 5px;
  font-size: 15px;
}

@media (min-width: 800px) {
  .user-list-card {
    font-size: 18px;
  }
}

.user-list-card a {
  color: blue;
  text-decoration-line: none;
}

.user-list-card a:hover {
  color: purple;
  text-decoration-line: underline;
}

.user-list-card a:active {
  color: aqual;
}
