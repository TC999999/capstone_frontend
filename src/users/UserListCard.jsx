import "../styles/UserListCard.css";
import { Link } from "react-router-dom";

const UserListCard = ({ user }) => {
  return (
    <div className="user-list-card">
      <h2>
        {" "}
        <Link to={`/users/${user.username}`}>{user.username}</Link>
      </h2>
      <ul>
        <li>
          <b>First Name: </b> {user.firstName}
        </li>
        <li>
          <b>Last Name: </b> {user.lastName}
        </li>
        <li>
          <b>Email Address: </b> {user.email}
        </li>
      </ul>
      <ul>
        <li>
          <b>Admin Status: </b> {user.isAdmin ? "Admin" : "User"}
        </li>
        <li>
          <b>Flag Status: </b> {user.isFlagged ? "Flagged" : "Not Flagged"}
        </li>
      </ul>
    </div>
  );
};

export default UserListCard;
