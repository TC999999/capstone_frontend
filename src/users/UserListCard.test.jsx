import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserProvider from "../UserProvider.jsx";
import { TEST_USER } from "../_testCommons.js";
import UserListCard from "./UserCard.jsx";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <UserListCard user={TEST_USER} />
      </UserProvider>
    </MemoryRouter>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <UserListCard user={TEST_USER} />
      </UserProvider>
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});
