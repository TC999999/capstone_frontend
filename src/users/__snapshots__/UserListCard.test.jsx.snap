// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="user-card"
  >
    <h2>
       
      <a
        href="/users/u1"
      >
        u1
      </a>
    </h2>
    <ul>
      <li>
        <b>
          Status: 
        </b>
         User
      </li>
      <li>
        <b>
          Average User Rating: 
        </b>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
